tests/Paginator/OffsetPaginatorTest.php

```diff
@@ -671,4 +671,83 @@ public function testReadOneWithLimit1(): void
 
         $this->assertSame(self::ITEM_1, $result);
     }
+
+    public function testNextPage(): void
+    {
+        $dataSet = [['id' => 1], ['id' => 2], ['id' => 3]];
+        $dataReader = new IterableDataReader($dataSet);
+        $paginator = (new OffsetPaginator($dataReader))->withPageSize(2);
+
+        // Test first page has next page
+        $nextPageReader = $paginator->nextPage();
+        $this->assertInstanceOf(OffsetPaginator::class, $nextPageReader);
+
+        // Verify the next page returns correct data
+        $nextPageData = array_values($this->iterableToArray($nextPageReader->read()));
+        $this->assertSame([['id' => 3]], $nextPageData);
+
+        // Test that returns null when there are no more pages
+        $this->assertNull($nextPageReader->nextPage());
+    }
+
+    public function testNextPageIterativeReading(): void
+    {
+        $dataSet = [['id' => 1], ['id' => 2], ['id' => 3]];
+        $sort = Sort::only(['id'])->withOrderString('id');
+        $dataReader = (new IterableDataReader($dataSet))->withSort($sort);
+        $paginator = (new OffsetPaginator($dataReader))->withPageSize(2);
+
+        $allData = [];
+        $currentPaginator = $paginator;
+
+        // Read all pages iteratively
+        while ($currentPaginator !== null) {
+            $pageData = array_values($this->iterableToArray($currentPaginator->read()));
+            $allData = array_merge($allData, $pageData);
+            $currentPaginator = $currentPaginator->nextPage();
+        }
+
+        // Verify we got all the data
+        $this->assertSame($dataSet, $allData);
+    }
+
+    public function testPreviousPage(): void
+    {
+        $dataSet = [['id' => 1], ['id' => 2], ['id' => 3]];
+        $sort = Sort::only(['id'])->withOrderString('id');
```

Comment:
```suggestion
        $sort = Sort::only(['id']);
```

Should work without `withOrderString('id')`.